climt/_core/initialization.py

```diff
@@ -561,74 +681,75 @@ def get_exponent_for_sigma(b_half, num_sigma_levels):
-    'heat_flux_into_sea_water_due_to_sea_ice': {'value': 0., 'units': 'W m^-2'},
-    'ocean_mixed_layer_thickness': {'value': 50., 'units': 'm'},
-    'soil_layer_thickness': {'value': 50., 'units': 'm'},
-    'heat_capacity_of_soil': {'value': 2000., 'units': 'J kg^-1 degK^-1'},
-    'area_type': {'value': 'sea', 'units': 'dimensionless', 'dtype': 'a100'},
-    'surface_upward_sensible_heat_flux': {'value': 0., 'units': 'W m^-2'},
-    'surface_upward_latent_heat_flux': {'value': 0., 'units': 'W m^-2'},
-    'land_ice_thickness': {'value': 0., 'units': 'm'},
-    'sea_ice_thickness': {'value': 0., 'units': 'm'},
-    'surface_snow_thickness': {'value': 0., 'units': 'm'},
+    'air_temperature': {'value': 290., 'units': 'degK', 'domain': 'atmosphere'},
+    'northward_wind': {'value': 0., 'units': 'm/s', 'domain': 'atmosphere'},
+    'eastward_wind': {'value': 0., 'units': 'm/s', 'domain': 'atmosphere'},
+    'divergence_of_wind': {'value': 0., 'units': 's^-1', 'domain': 'atmosphere'},
+    'atmosphere_relative_vorticity': {'value': 0., 'units': 's^-1', 'domain': 'atmosphere'},
+    'specific_humidity': {'value': 0., 'units': 'kg/kg', 'domain': 'atmosphere'},
+    'mole_fraction_of_carbon_dioxide_in_air': {'value': 330e-6, 'units': 'dimensionless', 'domain': 'atmosphere'},
+    'mole_fraction_of_methane_in_air': {'value': 0., 'units': 'dimensionless', 'domain': 'atmosphere'},
+    'mole_fraction_of_nitrous_oxide_in_air': {'value': 0., 'units': 'dimensionless', 'domain': 'atmosphere'},
+    'mole_fraction_of_oxygen_in_air': {'value': 0.21, 'units': 'dimensionless', 'domain': 'atmosphere'},
+    'mole_fraction_of_nitrogen_in_air': {'value': 0.78, 'units': 'dimensionless', 'domain': 'atmosphere'},
+    'mole_fraction_of_hydrogen_in_air': {'value': 500e-9, 'units': 'dimensionless', 'domain': 'atmosphere'},
+    'mole_fraction_of_cfc11_in_air': {'value': 0., 'units': 'dimensionless', 'domain': 'atmosphere'},
+    'mole_fraction_of_cfc12_in_air': {'value': 0., 'units': 'dimensionless', 'domain': 'atmosphere'},
+    'mole_fraction_of_cfc22_in_air': {'value': 0., 'units': 'dimensionless', 'domain': 'atmosphere'},
+    'mole_fraction_of_carbon_tetrachloride_in_air': {'value': 0., 'units': 'dimensionless', 'domain': 'atmosphere'},
+    'cloud_area_fraction_in_atmosphere_layer': {'value': 0., 'units': 'dimensionless', 'domain': 'atmosphere'},
+    'mass_content_of_cloud_ice_in_atmosphere_layer': {'value': 0., 'units': 'kg m^-2', 'domain': 'atmosphere'},
+    'mass_content_of_cloud_liquid_water_in_atmosphere_layer': {'value': 0., 'units': 'kg m^-2', 'domain': 'atmosphere'},
+    'cloud_ice_particle_size': {'value': 20., 'units': 'micrometer', 'domain': 'atmosphere'},
+    'cloud_water_droplet_radius': {'value': 10., 'units': 'micrometer', 'domain': 'atmosphere'},
+    'cloud_base_mass_flux': {'value': 0., 'units': 'kg m^-2 s^-1', 'domain': 'atmosphere_horizontal'},
+    'zenith_angle': {'value': 0., 'units': 'radians', 'domain': 'atmosphere_horizontal'},
```

Comment:
It is not uncommon to refer to the axis perpendicular to the earth's surface as the vertical, and the
surface of the earth to be referred to as horizontal. I think this name says exactly that -- a quantity that has the same shape as the grid that defines a horizontal atmospheric surface.

Reply:
If it were a "grid" and not a "domain, I would immediately think horizontal refers to a horizontal grid, but as it currently reads my mind goes to "A horizontal atmosphere? What? Is that like a slice of atmosphere across the horizontal?". If there are no better names then so be it.

This could be solved by this suggestion:

>You could actually determine the "domain" that a component wants a quantity to be on based on its dims for that quantity. [mid_levels, *], [mid_levels, latitude, longitude], etc. all map to atmosphere, for example.

Reply:
I'm fairly certain that I want to change many things about this init code, especially after our discussion yesterday. For now, I wanted something that would allow us to not skip tests that we disabled after the previous refactor.

Somewhere during the refactor, some tests have started failing which simply won't pass now, which is really bothering me. 

I will work on incorporating the ideas we discussed here over the coming days.

Reply:
These are tests which were "un-skipped" in the PR. they seem to point to issues with different compilers suddenly producing non-exact results for arbitrary (but deterministic) tests. This was not an issue earlier.

See https://travis-ci.org/CliMT/climt/jobs/437949736 for linux builds for example. What is strange is that not all tests for a component fail.

The issue is worse on appveyor. I'm trying to see if I can find an updated toolchain for Anaconda on Windows to eliminate this issue rather than playing around with NetCDF files.